cmake/mission_build.cmake

```diff
@@ -40,6 +40,11 @@ function(initialize_globals)
       set(MISSION_SOURCE_DIR ${MISSION_SOURCE_DIR} CACHE PATH "Top level mission source directory")
     endif(NOT DEFINED MISSION_SOURCE_DIR)
 
+    # Default core modules, set early so it can be appended or replaced from user configuration
+    if (NOT DEFINED MISSION_CORE_MODULES)
+      set(MISSION_CORE_MODULES msg CACHE PATH "Mission core modules")
+    endif (NOT DEFINED MISSION_CORE_MODULES)
+
```

Comment:
This shouldn't be a cache variable (it wasn't before, right?)

Cache is used for when you want to retain a user-specified config from run to run.  Values like this are read from a file (e.g. targets.cmake) so there is no need to cache - its in the file.  It can actually get rather weird when mixing cache and non-cache values, as they both coexist even with the same name - one hides the other - and behavior can be confusing when you WANT to change it i.e. by updating your config and rebuilding - if cache is used then the old value will continue to be active.

Reply:
Good point, let me know what you think of 67dbcc4